ThunderTable/TableViewController+Reload.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  TableViewController+Reload.swift
+//  ThunderTable
+//
+//  Created by Simon Mitchell on 17/07/2018.
+//  Copyright © 2018 3SidedCube. All rights reserved.
+//
+
+import UIKit
+
+extension TableViewController {
+    
+    /// Finds the index path for a particular row.
+    ///
+    /// - Note: The row in question must conform to Equatable for this to succeed.
+    ///
+    /// - Parameter row: The row to find the index path for.
+    /// - Returns: The index path the row is positioned at if it is in `data`.
+    public func indexPathFor<T: Row & Equatable>(row: T) -> IndexPath? {
+        
+        for (index, section) in data.enumerated() {
+            
+            guard let matchingRowIndex = section.rows.firstIndex(where: {
+                guard let matchableRow = $0 as? T else { return false }
+                return matchableRow == row
+            }) else {
+                continue
+            }
+            
+            return IndexPath(row: matchingRowIndex, section: index)
+        }
+        
+        return nil
+    }
+    
+    /// Replaces a row in `data` with another row, whilst optionally reloading other index paths.
+    ///
+    /// - Note: This relies on the containing section of the row being of class `TableSection` as apposed to any object conforming to `Section`
+    ///
+    /// - Parameters:
+    ///   - row: The row that should be replaced.
+    ///   - otherRow: The row that is replacing the original row.
+    ///   - additionalReloadIndexPaths: Additional index paths that should be reloaded at the same time.
+    ///   - animation: The animation to use when reloading the rows
+    public func replace<T: Row & Equatable>(row: T, with otherRow: Row, reloading additionalReloadIndexPaths: [IndexPath] = [], animation: UITableView.RowAnimation = .none) {
+        
+        guard let indexPath = indexPathFor(row: row) else { return }
+        guard let tableSection = data[indexPath.section] as? TableSection else { return }
+        
+        var rows = tableSection.rows
+        rows[indexPath.row] = otherRow
+        tableSection.rows = rows
+        
+        withoutRedrawing {
+            data[indexPath.section] = tableSection
+        }
+        
+        var indexPaths = [indexPath]
+        indexPaths.append(contentsOf: additionalReloadIndexPaths)
```

Comment:
Happy to leave, but user's could wrap `additionalReloadIndexPaths` logic into a `tableView.beginUpdates()` so it doesn't need to go via this method. But let's leave for now